File: backend/src/main/java/de/dlr/shepard/v2/template/io/CreateShepardTemplateIO.java
package de.dlr.shepard.v2.template.io;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

/**
 * POST body for {@code POST /v2/templates}.
 *
 * <p>{@code appId} is server-minted; {@code version} starts at 1.
 * {@code createdBy} comes from the authenticated principal — never
 * from the body.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(name = "CreateShepardTemplate")
public class CreateShepardTemplateIO {

  @Schema(required = true, description = "Human-readable template name.")
  private String name;

  @Schema(required = true, description = "Template kind, e.g. DATAOBJECT_RECIPE / COLLECTION_RECIPE / EXPERIMENT_RECIPE.")
  private String templateKind;

  @Schema(required = true, description = "Recipe body. JSON DSL per aidocs/54 §7.")
  private String body;

  @Schema(required = false, nullable = true, description = "Markdown-flavoured human description.")
  private String description;

  @Schema(required = false, nullable = true, description = "Author-supplied tags for picker filtering.")
  private List<String> tags;
}

File: backend-client/src/models/CreateShepardTemplateIO.ts
/* tslint:disable */
/* eslint-disable */

/**
 * POST/PATCH body for /v2/templates endpoints, per aidocs/54 §5.
 * Manually maintained; matches CreateShepardTemplateIO.java exactly.
 */
export interface CreateShepardTemplateIO {
  name: string;
  /** Template kind, e.g. DATAOBJECT_RECIPE / COLLECTION_RECIPE / EXPERIMENT_RECIPE */
  templateKind: string;
  /** Recipe body. JSON DSL per aidocs/54 §7 */
  body: string;
  description?: string | null;
  tags?: string[] | null;
  /**
   * Optional MDI icon name (e.g. "mdi-layers") — sets the template's
   * iconKey on create / patch. Empty string clears (UI falls back to
   * per-kind default). Design: aidocs/integrations/122.
   */
  iconKey?: string | null;
}
